### PR TITLE
Core: Enforce semicolons after various macros

### PR DESCRIPTION
**Pull request by Repiteo**
Implementation of semicolons after most macros has been inconsistent across the repo, but appears to be generally preferred stylistically. This PR aims to make the process uniform by taking several common macros & appending a new macro to the end of them: `_FORCE_SEMICOLON_`. This is a simple wrapper around `static_assert(true)` which has the benefit of mandating a semicolon after a macro WITHOUT compiling any extra code in the process. No associated `.git-blame-ignore-revs` commit, as there's different functionality despite the surface-level visual change.

### CODE REVIEW

File: core/object/make_virtuals.py [hunk 1]
```diff
@@ -185,9 +186,10 @@ def run(target, source, env):
 		tracker->initialized = &m_initialized;\\
 		tracker->next = virtual_method_list;\\
 		virtual_method_list = tracker;\\
-	}
+	}\\
+	_FORCE_SEMICOLON_
 #else
-#define GDVIRTUAL_TRACK(m_virtual, m_initialized)
+#define GDVIRTUAL_TRACK(m_virtual, m_initialized) static_assert()
```
Comment (AThousandShips) [inline core/object/make_virtuals.py:192]:
```suggestion
#define GDVIRTUAL_TRACK(m_virtual, m_initialized) _FORCE_SEMICOLON_
```

**Comment by AThousandShips** [comment]
I don't think it makes sense to enforce this for `GDCLASS`, sure it might look better but it's technically wrong, so I's say it's just adding a lot of noise for a minor visual detail

For the cases where the macro actually contains a statement that needs termination I'd say it's good to ensure it's correct, i.e. if the macro contains a semicolon already, to remove it, and when there's a disabled version of it to add enforcement there to make sure it's caught

But otherwise I think this just adds a ton of noise for entirely stylistic reasons, as there's no actual need to enforce it in those cases

**Comment by Repiteo** [comment]
Ohh, I'm not certain if there's a non-zero effect on compile time. Changing it to instead be a codestyle change would wholly eliminate potential ambiguity, which is indeed very different but that's not necessarily a bad thing. Perhaps it could take the form of a pre-commit hook check?

Gonna opt to make this a template to cut down on noise while I look into potential alternatives.

**Comment by AThousandShips** [comment]
I'm compiling now to check impact there

**Comment by Repiteo** [comment]
It seems my initial direction was misguided. Closing.

**Comment by lyuma** [comment]
@Repiteo 
Apologies for my wording. I wasn't expecting this issue to be closed without any debate. As many reviewers and interested parties, I am merely adding in a dissenting opinion to form the basis for debate: in many code style issues there is not one right answer, but there are arguments in each direction and sometimes a pushback to change.

When you replied suddenly that you "direction was misguided" in response to me merely expressing my *personal* opinions and preferences (I don't work for godot or anything), it makes me believe that I may have presented my argument in an inappropriate way.

I always like to consider both sides in a debate and your idea was not without merit of its own so I find this change of mind very rapid

I do hope my comment did not come off wrong, and I apologize.

Either way I wish you best and hope this won't discourage your future contributions. Even holding this debate here is a contribution: someone had to suggest the idea of semicolons after macros.

**Comment by Repiteo** [comment]
I appreciate you reaching out to clarify, but I promise that I took no offense to you words nor believed they came off in any coarse/rude manner. It arguably was rash to close this as quickly as I did, but I more attribute that to simply not considering other avenues and possibilities when making the PR in the first place. With that realization in mind, I wasn't interested in dying on this hill nor keeping up with rebasing until a concrete direction was formed.

With that all said, I'm not discouraged at all; on the contrary, I'm already brainstorming other PRs as we speak! This wasn't my first time jumping the gun with an idea and it won't be the last 😎

**Pull request closed without merging**